### ISSUE TITLE
onnx parsing issue due to Gather[axis=0] in onnx model converted from pytorch model

### ISSUE THREAD

**Issue by sunskyhsh** [pytorch/pytorch#28741]
## 🐛 Bug

Onnx can't parse the onnx model converted from pytorch '.pth' model, and reports following issue:
```
(py3.6_env) ➜  HRNet-Semantic-Segmentation git:(fd276ac) ✗ onnx2trt model/hrnet_w18_small_v1_cityscapes_cls19_1024x2048_trainset.onnx -o model/hrnet_w18_small_v1_cityscapes_cls19_1024x2048_trainset.trt
----------------------------------------------------------------
Input filename:   model/hrnet_w18_small_v1_cityscapes_cls19_1024x2048_trainset.onnx
ONNX IR version:  0.0.4
Opset version:    9
Producer name:    pytorch
Producer version: 1.3
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
Parsing model
WARNING: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
Successfully casted down to INT32.
While parsing node number 54 [Gather -> "490"]:
ERROR: /home/sunsky/repo/onnx-tensorrt/onnx2trt_utils.hpp:399 In function convert_axis:
[8] Assertion failed: axis >= 0 && axis < nbDims
```

This is how I convert pytorch model to onnx model.
```
def saveONNX(model, filepath):
    '''
    dummy_input = torch.randn(1, 3, 720, 1280, device='cuda')
    torch.onnx.export(model, dummy_input, filepath, verbose=True)
```

In the output log of above conversion  I find the Gather[axis=0] operations:
```
  %488 : Long() = onnx::Constant[value={3}](), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0]
  %489 : Tensor = onnx::Shape(%473), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0]
  %490 : Long() = onnx::Gather[axis=0](%489, %488), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0] # /home/sunsky/repo/HRNet-Semantic-Segmentation/tools/../lib/models/seg_hrnet.py:239:0
  %491 : Long() = onnx::Constant[value={2}](), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0]
  %492 : Tensor = onnx::Shape(%473), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0]
  %493 : Long() = onnx::Gather[axis=0](%492, %491), scope: HighResolutionNet/Sequential[stage2]/HighResolutionModule[0] # /home/sunsky/repo/HRNet-Semantic-Segmentation/tools/../lib/models/seg_hrnet.py:240:0
```

And the Gather[axis=0] operations come from the 239th and 240th line in seg_hrnet.py:
```
    def forward(self, x):
        if self.num_branches == 1:
            return [self.branches[0](x[0])]

        for i in range(self.num_branches):
            x[i] = self.branches[i](x[i])

        x_fuse = []
        for i in range(len(self.fuse_layers)):
            y = x[0] if i == 0 else self.fuse_layers[i][0](x[0])
            for j in range(1, self.num_branches):
                if i == j:
                    y = y + x[j]
                elif j > i:
                    width_output = x[i].shape[-1] # <==== THE 239th LINE
                    height_output = x[i].shape[-2] # <==== THE 240th LINE
                    y = y + F.interpolate(
                        self.fuse_layers[i][j](x[j]),
                        size=[height_output, width_output],
                        mode='bilinear')
                else:
                    y = y + self.fuse_layers[i][j](x[j])
            x_fuse.append(self.relu(y))

        return x_fuse
```

## To Reproduce

1. Git clone https://github.com/HRNet/HRNet-Semantic-Segmentation.git
1. Download a pretrained model from https://github.com/HRNet/HRNet-Semantic-Segmentation, [HRNetV2-W18-Small-v1](https://1drv.ms/u/s!Aus8VCZ_C_33gSEsg-2sxTmZL2AT?e=AqHbjh) for example.
1. Use torch.onnx.export to convert the pretrained model to onnx model.
1. Use [onnx-tensorrt](https://github.com/onnx/onnx-tensorrt) or TensorRT 5.1.5 to parse the onnx model.

## Environment

PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: version 3.14.4

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce RTX 2070
Nvidia driver version: 418.56
cuDNN version: 7.5.0.56

Versions of relevant libraries:
[pip3] numpy==1.16.4
[pip3] ros-numpy==0.0.2
[pip3] torch==1.3.0
[pip3] torchvision==0.4.1
[conda] torch                     1.3.0                     <pip>
[conda] torchvision               0.4.1                     <pip>

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by garymm** [comment]
@sunskyhsh is this still an issue with the latest nightly versions of PyTorch and TensorRT?

**Comment by thiagocrepaldi** [comment]
It’s been over 2 months, so closing this. Feel free to open another one.
